web/views/project_view.ex

```diff
@@ -38,6 +38,20 @@ defmodule Ask.ProjectView do
     }
   end
 
+  def render("activities.json", %{activities: activities}) do
+    %{data: render_many(activities, Ask.ProjectView, "activity.json", as: :activity)}
+  end
+
+  def render("activity.json", %{activity: activity}) do
+    %{user_name: (if activity.user, do: activity.user.name, else: activity.remote_ip),
```

Comment:
Let's add two separate fields, and decide on the UI if the remote IP must be displayed